### ISSUE TITLE
External Player not working with some files

### ISSUE THREAD

**Issue by harry71019** [jellyfin/jellyfin-android#192]
**Describe the bug**
External player fails to play for some files and shows "Invalid Play method" 

**To Reproduce**
1. Try to play certain file 
2. Playback shows error, external player not launched
3. Turn off transcoding for that user
4. Now playback works

**Expected behavior**
The file should be able to played without transcoding

**Logs**
This is the log when playback error occurs
[log2.txt](https://github.com/jellyfin/jellyfin-android/files/5390949/log2.txt)

This is the log playing the same file but turning off transcoding for that user
[log3.txt](https://github.com/jellyfin/jellyfin-android/files/5390952/log3.txt)

**System (please complete the following information):**
 - App version: 2.0.1
 - Android version: Android 9.0
 - Device manufacturer and model: HTC U11
 - Jellyfin server version: 10.6.4
 - The type of video player implementation used: External player

**Additional context**
[Keyakizaka46 - Keyaki Kyouwakoku 2018.chi.default.zip](https://github.com/jellyfin/jellyfin-android/files/5390957/Keyakizaka46.-.Keyaki.Kyouwakoku.2018.chi.default.zip)
Since the logs mention unsupported subtitles I'll include the subtitles here. But I have plenty other files with .ass subtitles working fine so I'm not sure what's causing the issue.

**Comment by CarlosOlivo** [comment]
Could you also share the metadata of the .mkv files? Preferably from 2 mkv with .ass subtitles, one that plays without problems and another that reports problems in the subtitles, You can use [MediaInfoOnline](https://mediaarea.net/MediaInfoOnline).

**Comment by harry71019** [comment]
@CarlosOlivo 
The file I just showed above was using external subtitles
folder structure:
Movie.mkv
Movie.chi.default.ass

Do I just show the mkv's mediainfo or should I add the mediainfo for the subtitles too

**Comment by Maxr1998** [comment]
> Do I just show the mkv's mediainfo or should I add the mediainfo for the subtitles too

Only for the mkv. On Linux, you can also use the `mediainfo` tool for that.

**Comment by harry71019** [comment]
Now that you mention it, I think the problem might be because of external subtitles?
The mediainfo for the file with problems:
[Keyakizaka46 - Keyaki Kyouwakoku 2018.mkv_MediaInfo.txt](https://github.com/jellyfin/jellyfin-android/files/5392194/Keyakizaka46.-.Keyaki.Kyouwakoku.2018.mkv_MediaInfo.txt)

The mediainfo for the file with ass sub but no problems (the ass is muxed in mkv)
[KEYAKIZAKA46 Live Online, but with YOU ! [ZAIKO].mkv_MediaInfo.txt](https://github.com/jellyfin/jellyfin-android/files/5392195/KEYAKIZAKA46.Live.Online.but.with.YOU.ZAIKO.mkv_MediaInfo.txt)

I also tried playing another movie with external srt subtitles, the movie can be played in external players, but the subtitle will not show up in the external player.

**Comment by CarlosOlivo** [comment]
> Now that you mention it, I think the problem might be because of external subtitles?

Bingo!

https://github.com/jellyfin/jellyfin-android/blob/148ea46a3fc51fb9c364914a06a9df0227b1ac58/app/src/main/assets/native/externalplayer.js#L193

https://github.com/jellyfin/jellyfin-android/blob/148ea46a3fc51fb9c364914a06a9df0227b1ac58/app/src/main/assets/native/externalplayer.js#L202

Adding 'ass' subtitle to externalSubtitleProfiles should fix this, maybe merge both lists?

> I also tried playing another movie with external srt subtitles, the movie can be played in external players, but the subtitle will not show up in the external player.

This was recently fixed in #177. I guess it will be included in one of the next releases.

**Comment by harry71019** [comment]
Nice, that looks promising, hopefully the next release comes out soon so I can test that.

**Comment by CarlosOlivo** [comment]
You can test now if you want, download [jellyfin-android.zip](https://dev.azure.com/jellyfin-project/7cce6c46-d610-45e3-9fb7-65a6bfd1b671/_apis/build/builds/16562/artifacts?artifactName=jellyfin-android&api-version=6.0&%24format=zip), extract and try to install `jellyfin-android-v0.0.0-dev.1-debug.apk` to see if #193 fixed this.

**Comment by harry71019** [comment]
I tested it with MX player / VLC / mpv-android, both MX player and mpv were able to read the external srt / ass subtitles.
I could play the video using VLC, but external subtitles will not get passed to VLC.

**Comment by CarlosOlivo** [comment]
Unfortunately VLC for Android does not allow external subtitles yet https://trac.videolan.org/vlc/ticket/18736

**Comment by harry71019** [comment]
Thanks for the info, just wanted to make sure it wasn't another weird bug or anything.
Fortunately the fix works well with MX player and mpv.

**Comment by blackCauldron7** [comment]
Not working on mpv-android. Says invalid play method when using subtitles.

**Comment by rluetzner** [comment]
Maybe this will help someone. I ran into the same issue that Jellyfin Android would tell me "invalid play method" for some files when using an external player (in my case VLC).

After messing around for a bit I discovered that this would only happen with files that were loaded with subtitles, i.e. when selecting a movie the subtitle box would show a language instead of "off".

In my case it helped to go into the settings --> subtitles and select subtitle mode "none". I can still turn on the subtitles in VLC after the file started playing. After this change I was able to use VLC consistently.

This is probably related to another bug I found where turning the subtitle off via the selection box does not register immediately and the movie has to be started once, stopped and restarted. I mainly noticed this because any preselected subtitles would force transcoding when played, wheras turning subtitles off will allow for direct play (where subtitles can be enabled at any later point). I could only observe this with the internal player. I guess the "invalid play method" could not be resolved because playback could never be started with the external player and the "off" setting for subtitles was never *confirmed*.
